.github/workflows/boefjes-tests.yml

```diff
@@ -29,11 +29,66 @@ jobs:
       - name: Install Octopoes
         run: cd octopoes && python setup.py bdist_wheel && pip install dist/octopoes*.whl
 
-      - name: Run pytests
-        run: python3 -m pytest --cov boefjes/ --cov-report xml --cov-branch boefjes/tests
+      - name: Run tests
+        run: python3 -m pytest --cov boefjes/ --cov-report xml --cov-branch tests/
+        working-directory: boefjes/
 
       - name: Upload coverage as artifact
         uses: actions/upload-artifact@v4
         with:
           name: boefjes-coverage-unit
-          path: coverage.xml
+          path: boefjes/.coverage
+          include-hidden-files: true
+
+  integration-tests:
+    runs-on: ubuntu-24.04
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Run integration tests
+        run: make itest
+        working-directory: boefjes/
+
+      - name: Upload coverage as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: boefjes-coverage-integration
+          path: boefjes/.coverage
+          include-hidden-files: true
+
+  coverage:
+    runs-on: ubuntu-24.04
+    needs:
+      - unit-tests
+      - integration-tests
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
```

Comment:
As far as I can see checking out is unnecessary, because merging the coverage files only uses the downloaded artifacts.

Reply:
I remember leaving this out in the first place, but I found out `coverage` needs the source files for mapping and reporting